src/Libraries/Microsoft.Extensions.AI.OpenAI/OpenAISerializationHelpers.cs

```diff
@@ -0,0 +1,100 @@
+    /// </summary>
+    /// <param name="stream">The stream containing a message using the OpenAI wire format.</param>
+    /// <param name="cancellationToken">A token used to cancel the operation.</param>
+    /// <returns>The deserialized list of chat messages and chat options.</returns>
+    public static async Task<OpenAIChatCompletionRequest> DeserializeChatCompletionRequestAsync(
+        Stream stream, CancellationToken cancellationToken = default)
+    {
+        _ = Throw.IfNull(stream);
+
+        BinaryData binaryData = await BinaryData.FromStreamAsync(stream, cancellationToken).ConfigureAwait(false);
+        ChatCompletionOptions openAiChatOptions = JsonModelHelpers.Deserialize<ChatCompletionOptions>(binaryData);
+        return OpenAIModelMappers.FromOpenAIChatCompletionRequest(openAiChatOptions);
+    }
+
+    /// <summary>
+    /// Serializes a Microsoft.Extensions.AI completion using the OpenAI wire format.
+    /// </summary>
+    /// <param name="stream">The stream to write the value.</param>
+    /// <param name="chatCompletion">The chat completion to serialize.</param>
+    /// <param name="options">The <see cref="JsonSerializerOptions"/> governing function call content serialization.</param>
+    /// <param name="cancellationToken">A token used to cancel the serialization operation.</param>
+    /// <returns>A task tracking the serialization operation.</returns>
+    public static async Task SerializeAsync(
+        Stream stream,
+        ChatCompletion chatCompletion,
+        JsonSerializerOptions? options = null,
+        CancellationToken cancellationToken = default)
+    {
+        _ = Throw.IfNull(stream);
+        _ = Throw.IfNull(chatCompletion);
+        options ??= AIJsonUtilities.DefaultOptions;
+
+        OpenAI.Chat.ChatCompletion openAiChatCompletion = OpenAIModelMappers.ToOpenAIChatCompletion(chatCompletion, options);
+        BinaryData binaryData = JsonModelHelpers.Serialize(openAiChatCompletion);
+#if NET
+        await stream.WriteAsync(binaryData.ToMemory(), cancellationToken).ConfigureAwait(false);
+#else
+        await stream.WriteAsync(binaryData.ToArray(), 0, binaryData.Length, cancellationToken).ConfigureAwait(false);
+#endif
+    }
+
+    /// <summary>
+    /// Serializes a Microsoft.Extensions.AI streaming completion using the OpenAI wire format.
+    /// </summary>
+    /// <param name="stream">The stream to write the value.</param>
+    /// <param name="streamingChatCompletionUpdates">The streaming chat completions to serialize.</param>
+    /// <param name="options">The <see cref="JsonSerializerOptions"/> governing function call content serialization.</param>
+    /// <param name="cancellationToken">A token used to cancel the serialization operation.</param>
+    /// <returns>A task tracking the serialization operation.</returns>
+    public static Task SerializeAsSseEventsAsync(
```

Comment:
SerializeStreamingAsync?

Reply:
I think specifying the format on the method name is important, given that all other methods in the class use pure JSON.

Reply:
The wire format details are irrelevant to the consumer of these APIs, though. They want the response written out as OpenAI would, for non-streaming or streaming. That streaming uses SSE is an implementation detail.

Reply:
Done.